tests/nodes/tests_Literal.cpp

```diff
@@ -457,3 +459,237 @@ TEST_CASE("Literal - casting") {
         CHECK(lit2.null());
     }
 }
+
+TEST_CASE("Literal - misc functions") {
+    using namespace rdf4cpp::rdf;
+
+    SUBCASE("rand") {
+        SUBCASE("same thread") {
+            auto const l1 = Literal::generate_random_double();
+            auto const l2 = Literal::generate_random_double();
+
+            CHECK(l1 >= 0.0_xsd_double);
+            CHECK(l1 < 1.0_xsd_double);
+
+            CHECK(l2 >= 0.0_xsd_double);
+            CHECK(l2 < 1.0_xsd_double);
+            CHECK(l1 != l2);
```

Comment:
please add a short comment that this check is non-deterministic. Will most probably never fail but should be documented.